pages/auth/signin.tsx

```diff
@@ -0,0 +1,129 @@
+import { signIn } from "next-auth/client";
+import Joi from "lib/validate";
+import React, { useEffect, useState } from "react";
+import { useForm } from "react-hook-form";
+import { joiResolver } from "@hookform/resolvers/joi";
+import Button from "components/Button";
+import { useRouter } from "next/router";
+import Link from "next/link";
+import FormField from "components/FormField";
+
+type UserSignInFormValues = {
+  email: string;
+  password: string;
+};
+
+const UserSignInFormSchema = Joi.object<UserSignInFormValues>({
+  email: Joi.string()
+    .trim()
+    .email({ tlds: { allow: false } }),
+  password: Joi.forbidden().required(),
+});
+
+const SignIn: React.FC = () => {
+  const router = useRouter();
+  const [submitting, setSubmitting] = useState(false);
+  const [error, setError] = useState("");
+  const [email, setEmail] = useState("");
+  const [revealPassword, setRevealPassword] = useState(false);
+  const { errors, register, handleSubmit } = useForm<UserSignInFormValues>({
+    resolver: joiResolver(UserSignInFormSchema),
+  });
+
+  useEffect(() => {
+    if (router.query.error) {
+      setError(router.query.error as string);
+    }
+    if (router.query.email) {
+      setEmail(router.query.email as string);
+    }
+  }, [router.query.error, router.query.email]);
+
+  async function onSubmit(
+    values: UserSignInFormValues,
+    event?: React.BaseSyntheticEvent
+  ): Promise<void> {
+    event?.preventDefault();
+    if (submitting) {
+      return;
+    }
+    try {
+      await signIn("credentials", {
+        email: values.email,
+        password: values.password,
+        callbackUrl: `${window.location.origin}/admin/players`,
```

Comment:
change "admin" to whatever type of user is signed in — you can usually do this by using useSession — though I am not sure if you can do that here with nextAuth but try it out! if not lmk :)